op-e2e/system_test.go

```diff
@@ -37,13 +46,55 @@ import (
 	oppprof "github.com/ethereum-optimism/optimism/op-service/pprof"
 )
 
+func TestMain(m *testing.M) {
+	flag.BoolVar(&verboseGethNodes, "gethlogs", true, "Enable logs on geth nodes")
+	flag.StringVar(&externalL2Nodes, "externalL2", "", "Enable tests with erigon")
+	flag.Parse()
+
+	if externalL2Nodes != "" {
+		fmt.Println("Running tests with external L2 process adapter at ", externalL2Nodes)
+		shimPath, err := filepath.Abs(externalL2Nodes)
+		if err != nil {
+			fmt.Printf("Could not compute abs of externalL2Nodes shim: %s\n", err)
+			os.Exit(2)
+		}
+		// We convert the passed in path to an absolute path, as it simplifies
+		// the path handling logic for the rest of the testing
+		externalL2Nodes = shimPath
+
+		_, err = os.Stat(externalL2Nodes)
+		if err != nil {
+			fmt.Printf("Failed to stat externalL2Nodes path: %s\n", err)
+			os.Exit(3)
+		}
+
+		cmd := exec.Command(externalL2Nodes, "--init")
+		cmd.Dir = filepath.Dir(shimPath)
```

Comment:
I think making it a pre-requisite that the client is built avoids this uncertainty as it can be built into a normal `bin` directory just like all the other executables.  That's the approach we're using for things like `op-program` and `cannon` which are executed as part of e2e tests in separate processes.